backend/typescript/services/interfaces/IReviewService.ts

```diff
@@ -70,7 +76,8 @@ export interface ReviewResponseDTO {
   body: string;
   byline: string;
   featured: boolean;
-  createdBy: number;
+  createdBy?: number | null;
+  createdByUser?: User | null;
```

Comment:
Is there a reason we are returning both the User object and the createdBy number (which I am assuming is the user id?) here? Looking at the notion ticket, I think the sample response only includes the createdByUser object:
`HTTP 200
{
	"id": x,
	...
	"created_by_user": {
		"id": user id
		"name": admin name
	},
	...
}`

I think returning only the userId is the more restful way of doing it, but if we stick with the user object, not sure why we need both fields here

Reply:
If you take a look at the logic in the /GET and /POST endpoints, you'll see that `createdBy` is returned in the /POST request and `createdByUser` is returned in the /GET request. Since both request handlers were using the same DTO, I used this workaround to return one or the other.

Reply:
Oh right that makes sense. Minor concern but as far as I know, the more restful way of doing this would be to only return the ID in both cases and if the user needs the user data, they would make another request. I don't think it is worth the changes needed outside the scope of this ticket if we make this change though so I am cool with it

Reply:
cc @susheelk as this was in the ticket requirements

Reply:
@dhmi19 ordinarily I would agree, we should only return the exact data we need and just ids for relationships, but a specific user requirement when displaying a review is to show the name of its creator. (ie we'd never display a review without showing who wrote it). I thought it not worth always sending two requests here

Reply:
makes sense! I am good with it